### PR TITLE
Rename wallet *Controller keyed services to *Service

### PR DESCRIPTION
**Pull request by yrliou**
Resolves https://github.com/brave/brave-browser/issues/20243

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by bbondy — APPROVED** [review]
Since this is just a huge renaming I just reviewed the mojom file and made sure build files and .ts files are updated too. If CI passes and things seem to work I'm good.

**Review by kylehickinson — APPROVED** [review]
Went over a bunch of the iOS files and double checked, everything seems to look good! 👍 Will wait to make sure that iOS CI passes

**Review by mkarolin — APPROVED** [review]
`chromium_src` change LGTM

File: components/brave_wallet/browser/json_rpc_service.cc [hunk 1]
```diff
@@ -42,9 +42,9 @@ constexpr char kDomainPattern[] =
 net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
   return net::DefineNetworkTrafficAnnotation("json_rpc_service", R"(
       semantics {
-        sender: "ETH JSON RPC Controller"
+        sender: "ETH JSON RPC Service"
```
Comment (bbondy) [inline components/brave_wallet/browser/json_rpc_service.cc:45]: nit: Remove ETH here

File: browser/brave_wallet/BUILD.gn [hunk 2]
```diff
@@ -18,8 +18,8 @@ source_set("brave_wallet") {
     "eth_tx_service_factory.h",
     "keyring_service_factory.cc",
     "keyring_service_factory.h",
-    "rpc_controller_factory.cc",
-    "rpc_controller_factory.h",
+    "json_rpc_service_factory.cc",
+    "json_rpc_service_factory.h",
```
Comment (bbondy) [inline browser/brave_wallet/BUILD.gn:20]: Thanks for making these consistent, they bothered me for a while now :)

File: ios/browser/brave_wallet/asset_ratio_service_factory.cc [hunk 3]
```diff
@@ -24,7 +24,7 @@ mojom::AssetRatioService* AssetRatioServiceFactory::GetForBrowserState(
 }
 
 // static
-AssetRatioService* AssetRatioServiceFactory::GetControllerForBrowserState(
+AssetRatioService* AssetRatioServiceFactory::GetServiceForState(
```
Comment (yrliou) [inline ios/browser/brave_wallet/asset_ratio_service_factory.cc:27]: Can't use `GetServiceForBrowserState` due to bellow build error:
```
[2022-01-04T22:01:14.972Z] ../../brave/ios/browser/brave_wallet/asset_ratio_service_factory.cc:23:63: error: too many arguments to function call, expected single argument 'browser_state', have 2 arguments
[2022-01-04T22:01:14.972Z]       GetInstance()->GetServiceForBrowserState(browser_state, true));
[2022-01-04T22:01:14.973Z]       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                ^~~~
[2022-01-04T22:01:14.973Z] ../../brave/ios/browser/brave_wallet/asset_ratio_service_factory.h:31:29: note: 'GetServiceForBrowserState' declared here
[2022-01-04T22:01:14.973Z]   static AssetRatioService* GetServiceForBrowserState(
[2022-01-04T22:01:14.973Z]                             ^
[2022-01-04T22:01:14.973Z] ../../brave/ios/browser/brave_wallet/asset_ratio_service_factory.cc:30:63: error: too many arguments to function call, expected single argument 'browser_state', have 2 arguments
[2022-01-04T22:01:14.973Z]       GetInstance()->GetServiceForBrowserState(browser_state, true));
[2022-01-04T22:01:14.973Z]       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                ^~~~
[2022-01-04T22:01:14.974Z] ../../brave/ios/browser/brave_wallet/asset_ratio_service_factory.cc:27:46: note: 'GetServiceForBrowserState' declared here
[2022-01-04T22:01:14.974Z] AssetRatioService* AssetRatioServiceFactory::GetServiceForBrowserState(
[2022-01-04T22:01:14.974Z]                                              ^
[2022-01-04T22:01:14.974Z] 2 errors generated.
```

**Review by Douglashdaniel — APPROVED** [review]
`ts` and `tsx` files look good!

**Review by iefremov — APPROVED** [review]
browser/net/* LGTM

**Comment by bbondy** [comment]
Went through all the changes more thoroughly and they look good to me

**Pull request merged**